src/detect-content.c

```diff
@@ -379,20 +379,29 @@ _Bool DetectContentPMATCHValidateCallback(const Signature *s)
     if (!(s->flags & SIG_FLAG_DSIZE)) {
         return TRUE;
     }
-
     int max_right_edge_i = SigParseGetMaxDsize(s);
     if (max_right_edge_i < 0) {
         return TRUE;
     }
 
     uint32_t max_right_edge = (uint32_t)max_right_edge_i;
+    int content_temp_length = 0;
```

Comment:
can this not be declared on first use inside the for loop?

Reply:
Looks like I'm wrong here. I see now it's being used to store something across loop iterations. So it can stay here.